cpp/src/arrow/compute/kernels/aggregate_basic_internal.h

```diff
@@ -1002,6 +1115,11 @@ struct MinMaxInitState {
     return Status::OK();
   }
 
+  Status Visit(const DictionaryType&) {
+    state.reset(new DictionaryMinMaxImpl<SimdLevel>(out_type, options));
```

Comment:
In DictionaryMinMaxImpl there are many function calls on `input.value_type` such as 'MinMax' in `Consume` and 'greater'&'less' in `CompareMinMax`. Each function call goes through a dispatching process and some other function call overheads. Since the value type is fixed, we can get its corresponding kernels using `Function::DispatchExact` here and directly invoke the kernels in DictionaryMinMaxImpl.

This is also helpful when user inputs an unsupported type such as `dict(list(int32))`, we can fail early and return a better error message if we try dispatching here. Currently it will first compact the dictionary before failing and the error message is something like `MinMax has no kernel matching list(int32)` instead of showing the real input type `dict(list(int32))`.

But this can be left as an optimization for the future. You can create an issue and leave this PR as it is.

Reply:
Good insight. I tend to leave it as an new issue.

Reply:
Good idea, this could reduce the time that calling `MinMax`